packages/components/src/InputFile/InputFile.test.tsx

```diff
@@ -175,6 +175,29 @@ describe("Post Requests", () => {
         );
       });
     });
+
+    it("calls onError callback with the passed error if getUploadParams throws an error", async () => {
+      const fetchParams = jest.fn(() =>
+        Promise.reject(new Error("Custom Error Message")),
+      );
+      const handleError = jest.fn();
+
+      const { container } = render(
+        <InputFile getUploadParams={fetchParams} onUploadError={handleError} />,
+      );
+
+      const input = container.querySelector(
+        "input[type=file]",
+      ) as HTMLInputElement;
```

Comment:
I know that this is how we select this element in all other tests in this file, however, it is recommended by RTL to use semantic queries whenever possible. It doesn't seem like we have a `role` or any other accessible attributes on the input itself, which isn’t inherently a11y‑bad here because users interact with the visible `Button` (`role=button`, proper `name`) to open the file picker.
Still I would suggest using a more semantic query, like `getByTestId` here (also [suggested by RTL](https://testing-library.com/docs/queries/about/#manual-queries)):

```suggestion
      const input = screen.getByTestId("input-file-input");
```

You will then be able to remove the need to destructure `container` on line 185.

If you don't want to change other instances of querying this input in this test file, that is understandable and acceptable. But I believe that using a more recommended approach in the new test would be preferable.

Reply:
@nad182 [26726eb](https://github.com/GetJobber/atlantis/pull/2832/commits/26726eb61dc48c05063df3a23b7b823d0fbb89c5) cleans up all the tests where we were doing something like `container.querySelector("input[type=file]")` and replaced it with the suggested `screen.getByTestId` call instead.

There's still a series of tests in here that destructure `container` and compare it to a snapshot that I'm going to beg forgiveness and not try to deal with today, if that's cool with you.